python/lsst/cbp/coordUtils.py

```diff
@@ -0,0 +1,401 @@
+    return sorted(_ErrorList, key=lambda elt: elt[0])
+
+
+RefSpherePoint = SpherePoint(0, 0, radians)
+
+
+def getFlippedPos(xyPos, flipX):
+    """Get a 2-dimensional position with the x axis properly flipped
+
+    Parameters
+    ----------
+    xyPos : pair of `float`
+        Position to rotate
+    flipX : `bool`
+        True if the x axis should be flipped (negated)
+
+    Returns
+    -------
+    xyresult : pair of `float`
+        ``xyPos`` with the x axis flipped or not, according to ``flipX``
+    """
+    return (-xyPos[0], xyPos[1]) if flipX else xyPos
+
+
+def fieldAngleToVector(xyrad, flipX):
+    """Convert a pupil field angle to a pupil unit vector
+
+    Parameters
+    ----------
+    xyrad : `tuple` of 2 `float`
+        x,y :ref:`pupil field angle <lsst.cbp.pupil_field_angle>` (radians)
+
+    Returns
+    -------
+    vector : a numpy array of 3 `float`
+        A unit vector in the :ref:`pupil frame <lsst.cbp.pupil_frame>`
+    """
+    assert len(xyrad) == 2
+    xyradRightHanded = getFlippedPos(xyrad, flipX=flipX)
+    amount = math.hypot(*xyradRightHanded)*radians
+    bearing = math.atan2(xyradRightHanded[1], xyradRightHanded[0])*radians
+    return np.array(RefSpherePoint.offset(bearing=bearing, amount=amount).getVector())
+
+
+def vectorToFieldAngle(vec, flipX):
+    """Convert a vector to a pupil field angle
+
+    Parameters
+    ----------
+    vec : `iterable` of 3 `float`
+        3-dimensional vector in the
+        :ref:`pupil frame <lsst.cbp.pupil_frame>`;
+        the magnitude is ignored, but must be large enough
+        to compute an accurate unit vector
+    flipX : bool
+        Set True if the x axis of the focal plane is flipped
+        with respect to the pupil
+
+    Returns
+    -------
+    fieldAngle : `tuple` of 2 `float`
+        x,y :ref:`pupil field angle <lsst.cbp.pupil_field_angle>` (radians)
+    """
+    sp = SpherePoint(Vector3d(*vec))
+    amountRad = RefSpherePoint.separation(sp).asRadians()
+    bearingRad = RefSpherePoint.bearingTo(sp).asRadians()
+    xyrad = (amountRad*math.cos(bearingRad),
+             amountRad*math.sin(bearingRad))
+    return getFlippedPos(xyrad, flipX=flipX)
+
+
+def pupilPositionToVector(xyPos, flipX):
+    """Convert a pupil plane position to a 3D vector
+
+    Parameters
+    ----------
+    xyPos : sequence of 2 `float`
+        :ref:`pupil plane position <lsst.cbp.pupil_position>` (mm)
+    flipX : `bool`
+        True if the x axis of the position flipped (negated)
+
+    Returns
+    -------
+    vector : sequence of 3 `float`
+        3-dimensional vector in the
+        :ref:`pupil frame <lsst.cbp.pupil_frame>`:
+        x = 0, y = plane position x, z = plane position y
+    """
+    xyPosRightHanded = getFlippedPos(xyPos, flipX=flipX)
+    return np.array([0, xyPosRightHanded[0], xyPosRightHanded[1]])
+
+
+def computeShiftedPlanePos(planePos, fieldAngle, shift):
+    """Compute the plane position of a vector on a plane
```

Comment:
Docstrings should not exceed column 79.

Reply:
Yeah, this came up in conversation. I didn't realise there were different rules for the docstrings. I need to go make a *lot* of changes to my code...

Reply:
(Please don't spend time running off and making a lot of changes to your code. Someday, we might automate that; otherwise, we'll schedule effort to do it at the appropriate juncture.)

Reply:
I agree with @jdswinbank and would point out that the style guide said nothing about 79 characters until a few days ago (and I'm not sure it says anything now) -- we had the RFC but it had not been implemented.

Reply:
I will file a community post soon. There is no plan to do a major big bang edit of all docstrings.

Reply:
Well, I won't write any _more_ that are over that length, but the code isn't merged yet, so it's somewhat debatable as to whether that's "new" or not... It's pushed though, so I'd argue it's not 🙂